src/java.base/share/classes/java/io/Console.java

```diff
@@ -145,6 +146,124 @@ public Reader reader() {
         throw newUnsupportedOperationException();
     }
 
+    /**
+     * Writes a string representation of the specified object to this console's
+     * output stream, terminates the line and then flushes the console.
+     *
+     * <p> The effects of {@code con.println(obj)} are as if by calling
+     * {@snippet lang = java:
+     *     con.format("%s%n", obj)
+     * }
+     *
+     * @param  obj
+     *         An object whose string representation is to write.
+     *
+     * @return  This console
+     *
+     * @since 23
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.IMPLICIT_CLASSES)
+    public Console println(Object obj) {
+        throw newUnsupportedOperationException();
+    }
+
+    /**
+     * Writes a string representation of the specified object to this console's
+     * output stream and then flushes the console.
+     *
+     * <p> The effects of {@code con.print(obj)} are as if by calling
+     * {@snippet lang = java:
+     *     con.format("%s", obj)
+     * }
+     *
+     * @param  obj
+     *         An object whose string representation is to write.
+     *
+     * @return  This console
+     *
+     * @since 23
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.IMPLICIT_CLASSES)
+    public Console print(Object obj) {
+        throw newUnsupportedOperationException();
+    }
+
+    /**
+     * Provides a prompt, then reads a single line of text from the console.
+     *
+     * @param  prompt
+     *         A prompt string.
+     *
+     * @throws IOError
+     *         If an I/O error occurs.
+     *
+     * @return  A string containing the line read from the console, not
+     *          including any line-termination characters, or {@code null}
+     *          if an end of stream has been reached without having read
+     *          any characters.
+     *
+     * @since 23
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.IMPLICIT_CLASSES)
+    public String input(String prompt) {
+        throw newUnsupportedOperationException();
+    }
+
+    /**
+     * Provides access to console for Implicitly Declared Classes.
+     *
```

Comment:
Would be good to mention here that this class consists entirely of static methods that are adapters for the corresponding instance methods on the Console instance. Also have blanket statement here that if System.console() returns null, these methods all throw IOError.

Reply:
Okay, will do. Do you want me to remove all mentions of `IOError` from the individual methods then?

Reply:
We agreed that it's probably OK to leave the `@throws` clause on each individual method's spec.